### PR TITLE
VIH-9560 allocated to is visible when feature flag is off 

### PR DESCRIPTION
**Pull request by abbasiTahir**
fixed the issue when it was showing the allocated to when feature flag was off . added the unit tests .

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/VIH-9560

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/bookings-list/bookings-list.component.spec.ts [hunk 1]
```diff
@@ -1289,5 +1290,21 @@ describe('BookingsListComponent', () => {
             component.onChangeNoAllocated();
             expect(bookingPersistService.selectedUsers.length).toEqual(count);
         });
+
+        it('should not show allocated to if work allocation feature flat is off', async () => {
```
Comment (will-craig) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/bookings-list/bookings-list.component.spec.ts:1294]: Spelling mistake: feature **flag**

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.spec.ts [hunk 2]
```diff
@@ -142,6 +155,23 @@ describe('HearingDetailsComponent', () => {
         const result = component.getParticipantInfo('123-1234');
         expect(result).toBe('');
     });
+    fdescribe('feature flag turn on and off ', () => {
+        fit('should not show allocated to if work allocation feature flat is off', async () => {
```
Comment (will-craig) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.spec.ts:159]: remove focus fdescribe and fit -> set it back to describe it

**Review by will-craig — CHANGES_REQUESTED** [review]

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/bookings-list/bookings-list.component.spec.ts [hunk 3]
```diff
@@ -1289,5 +1290,21 @@ describe('BookingsListComponent', () => {
...
+            launchDarklyServiceSpy.flagChange.next({ 'vho-work-allocation': false });
+            await component.ngOnInit();
+            fixture.detectChanges();
+            const divTohide = fixture.debugElement.query(By.css('allocated-to'));
```
Comment (will-craig) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/bookings-list/bookings-list.component.spec.ts:1298]: variables should be camel-case divToHide

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/bookings-list/bookings-list.component.spec.ts [hunk 4]
```diff
@@ -1289,5 +1290,21 @@ describe('BookingsListComponent', () => {
...
+            expect(divTohide).toBeFalsy();
+        });
+
+        it('should  show allocated to if work allocation feature flat is off', async () => {
```
Comment (will-craig) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/bookings-list/bookings-list.component.spec.ts:1302]: double space between should _ _show

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.spec.ts [hunk 5]
```diff
@@ -142,6 +155,23 @@ describe('HearingDetailsComponent', () => {
...
+            launchDarklyServiceSpy.flagChange.next({ 'vho-work-allocation': false });
+            await component.ngOnInit();
+            fixture.detectChanges();
+            const divTohide = fixture.debugElement.query(By.css('hearing-allocated-to'));
+            expect(divTohide).toBeFalsy();
+        });
+
+        it('should  show allocated to if work allocation feature flat is on', async () => {
```
Comment (will-craig) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.spec.ts:167]: double space between should _ _show
Spelling mistake: feature **flag**

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.spec.ts [hunk 6]
```diff
@@ -181,31 +211,32 @@ describe('HearingDetailsComponent join by phone', () => {
         '1234567'
     );
     it('should display option to join by phone if config has not the set date', () => {
-        const component = new HearingDetailsComponent(activatedRoute, loggerSpy, configServiceSpy);
+        // @ts-ignore
```
Comment (will-craig) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.spec.ts:214]: Why have you added @ts-ignore ?

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.spec.ts [hunk 7]
```diff
@@ -11,9 +11,17 @@ import { ClientSettingsResponse } from 'src/app/services/clients/api-client';
 import { Logger } from '../../services/logger';
 import { OtherInformationModel } from '../../common/model/other-information.model';
 import { ConfigService } from 'src/app/services/config.service';
-import { of } from 'rxjs';
+import { of, ReplaySubject } from 'rxjs';
+import { LaunchDarklyService } from '../../services/launch-darkly.service';
+import { ReturnUrlService } from '../../services/return-url.service';
+import { FeatureFlagService } from '../../services/feature-flag.service';
```
Comment (will-craig) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.spec.ts:17]: This import isnt used

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.ts [hunk 8]
```diff
@@ -1,26 +1,51 @@
...
+    constructor(
+        private route: ActivatedRoute,
+        private logger: Logger,
+        private configService: ConfigService,
+        private lanchDarklyService: LaunchDarklyService
+    ) {}
+
+    ngOnInit() {
+        this.$ldSubcription = this.lanchDarklyService.flagChange.subscribe(value => {
+            if (value) {
```
Comment (will-craig) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.ts:44]: Not sure we need this if (value) condtion

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/bookings-list/bookings-list.component.spec.ts (hunk 1 above)
Comment (abbasiTahir) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/bookings-list/bookings-list.component.spec.ts:1294]: resolved changed .

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.spec.ts (hunk 2 above)
Comment (abbasiTahir) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.spec.ts:159]: changed the fdescribe.

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/bookings-list/bookings-list.component.spec.ts (hunk 3 above)
Comment (abbasiTahir) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/bookings-list/bookings-list.component.spec.ts:1298]: made the changes

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/bookings-list/bookings-list.component.spec.ts (hunk 4 above)
Comment (abbasiTahir) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/bookings-list/bookings-list.component.spec.ts:1302]: made the required changes .

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.spec.ts (hunk 5 above)
Comment (abbasiTahir) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.spec.ts:167]: changed.

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.spec.ts (hunk 6 above)
Comment (abbasiTahir) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.spec.ts:214]: may be automated not added by myself.

Comment (abbasiTahir) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.spec.ts]: removed ts ignore

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.spec.ts (hunk 7 above)
Comment (abbasiTahir) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.spec.ts:17]: rmeoved  the import .

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.ts (hunk 8 above)
Comment (abbasiTahir) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.ts:44]: removed the if condition

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.spec.ts [hunk 9]
```diff
@@ -142,6 +155,23 @@ describe('HearingDetailsComponent', () => {
         const result = component.getParticipantInfo('123-1234');
         expect(result).toBe('');
     });
+    describe('feature flag turn on and off ', () => {
+        fit('should not show allocated to if work allocation feature flag is off', async () => {
```
Comment (will-craig) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.spec.ts:159]: Still havent remove this 'fit'

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.spec.ts (hunk 2 above)
Comment (will-craig) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.spec.ts]: Not quite.. fit aswell

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/bookings-list/bookings-list.component.spec.ts (hunk 4 above)
Comment (will-craig) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/bookings-list/bookings-list.component.spec.ts]: still a double space betwen should and show

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.spec.ts (hunk 5 above)
Comment (will-craig) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.spec.ts]: still a double space betwen should and show

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.spec.ts (hunk 2 above)
Comment (abbasiTahir) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.spec.ts]: fixed

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.spec.ts (hunk 9 above)
Comment (abbasiTahir) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/hearing-details/hearing-details.component.spec.ts]: removed the fit

**Review by will-craig — APPROVED** [review]

**Review by will-craig — APPROVED** [review]

**Pull request merged**